src/store/models/modals.ts

```diff
@@ -235,6 +253,102 @@ const modalsModel: ModalsModel = {
       defaultCommand: undefined,
     });
   }),
+  setBalanceChannels: action((state, payload) => {
+    state.balanceChannels = {
+      ...state.balanceChannels,
+      ...payload,
+    };
+  }),
+  showBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: true });
+  }),
+  hideBalanceChannels: thunk(actions => {
+    actions.setBalanceChannels({ visible: false });
+  }),
+  resetBalanceChannels: thunk(
+    async (actions, network, { getStoreActions, getStoreState }) => {
+      const channels = [] as LightningNodeChannel[];
+      const { getChannels } = getStoreActions().lightning;
+      const { links } = getStoreState().designer.activeChart;
+
+      const id2Node = {} as Record<string, LightningNode>;
+      const promisesToAwait = [] as Promise<unknown>[];
+      const channelsInfo = [] as ChannelInfo[];
+
+      for (const node of network.nodes.lightning) {
+        promisesToAwait.push(
+          getChannels(node).then((nodeChannels: LightningNodeChannel[]) => {
+            channels.push(...nodeChannels);
+            id2Node[node.name] = node;
+          }),
+        );
+      }
+      await Promise.all(promisesToAwait);
+
+      for (const channel of channels) {
+        const { uniqueId: id, localBalance, remoteBalance } = channel;
+        if (!links[id]) continue;
+        const from = links[id].from.nodeId;
+        const to = links[id].to.nodeId as string;
```

Comment:
This cast is dangerous because the `to.nodeId` could be `undefined`. It would be better to `continue` in this case.